src/elf.rs

```diff
@@ -5,8 +5,10 @@ use object_trait::Object;
 pub struct Elf<'a>(xmas_elf::ElfFile<'a>);
 
 impl<'a> Object<'a> for Elf<'a> {
-    fn parse(input: &'a [u8]) -> Elf<'a> {
-        Elf(xmas_elf::ElfFile::new(input))
+    type Error = ();
+
+    fn parse(input: &'a [u8]) -> Result<Elf<'a>, Self::Error> {
+        Ok(Elf(xmas_elf::ElfFile::new(input)))
```

Comment:
Yeah. `goblin` looks like it could be promising eventually, but still doesn't really have mach-o support. Would be nice to not have to even support this crate at all...